pallets/investments/src/lib.rs

```diff
@@ -708,190 +634,218 @@ where
-		InvestOrders::<T>::try_mutate(
-			&who,
-			investment_id,
-			|maybe_order| -> DispatchResultWithPostInfo {
-				let order = if let Some(order) = maybe_order.as_mut() {
-					order
-				} else {
-					Self::deposit_event(Event::<T>::InvestCollectedWithoutActivePosition {
-						who: who.clone(),
-						investment_id,
-					});
-					// TODO: Return correct weight
-					//       - Accountant::info() + Storage::read() + Storage::write()
-					return Ok(().into());
-				};
-				let mut collection = InvestCollection::<T::Amount>::from_order(order);
-				let mut collected_ids = Vec::new();
-				let cur_order_id = InvestOrderId::<T>::get(investment_id);
-				let last_processed_order_id = min(
-					order
-						.submitted_at()
-						.saturating_add(T::MaxOutstandingCollects::get()),
-					cur_order_id,
-				);
-
-				// The current order is not in processing
-				if order.submitted_at() == cur_order_id {
-					Self::deposit_event(Event::<T>::InvestCollectedForNonClearedOrderId {
-						who: who.clone(),
-						investment_id,
-					});
-					// TODO: Return correct weight
-					//       - Accountant::info() + 2 * Storage::read() + Storage::write()
-					return Ok(().into());
-				}
-
-				for order_id in order.submitted_at()..last_processed_order_id {
-					let fulfillment = ClearedInvestOrders::<T>::try_get(investment_id, order_id)
-						.map_err(|_| Error::<T>::OrderNotCleared)?;
+		InvestOrders::<T>::try_mutate(&who, investment_id, |maybe_order| {
+			// Exit early if order does not exist
+			let order = if let Some(order) = maybe_order.as_mut() {
+				order
+			} else {
+				Self::deposit_event(Event::<T>::InvestCollectedWithoutActivePosition {
+					who: who.clone(),
+					investment_id,
+				});
+				// TODO: Return correct weight
+				//       - Accountant::info() + Storage::read() + Storage::write()
+				return Ok((Default::default(), ().into()));
+			};
+
+			let mut collection = InvestCollection::<T::Amount>::from_order(order);
+			let mut collected_ids = Vec::new();
+			let cur_order_id = InvestOrderId::<T>::get(investment_id);
+			let last_processed_order_id = min(
+				order
+					.submitted_at()
+					.saturating_add(T::MaxOutstandingCollects::get()),
+				cur_order_id,
+			);
+
+			// Exit early if the current order is not in processing
+			if order.submitted_at() == cur_order_id {
+				Self::deposit_event(Event::<T>::InvestCollectedForNonClearedOrderId {
+					who: who.clone(),
+					investment_id,
+				});
+				// TODO: Return correct weight
+				//       - Accountant::info() + 2 * Storage::read() + Storage::write()
+				return Ok((Default::default(), ().into()));
+			}
 
+			let mut amount_payment: T::Amount = Default::default();
+			for order_id in order.submitted_at()..last_processed_order_id {
+				let fulfillment = ClearedInvestOrders::<T>::try_get(investment_id, order_id)
+					.map_err(|_| Error::<T>::OrderNotCleared)?;
+
+				let currency_payout =
```

Comment:
NOTE: Apart from changing the return result type of `do_collect_invest` from `()` to `CollectedAmounts`, this is the only other change. Here we denote the amount of pool currency which was converted into tranche tokens required for the `ExecutedCollectInvest` message.

Reply:
Based on [this DAO slack message](https://centrifugedao.slack.com/archives/C04HCK98KLL/p1692046938307889?thread_ts=1691760753.515679&cid=C04HCK98KLL), the amount converted amounts (pool currency for `CollectInvest` and tranche tokens for `CollectRedeem`) are still required.